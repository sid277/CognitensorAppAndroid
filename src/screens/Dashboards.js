import React, { useEffect, useReducer } from 'react';
import { Text, StyleSheet, FlatList, ActivityIndicator } from 'react-native';
import { apiStateReducer } from '../reducers/ApiStateReducer';
import CognitensorEndpoints from '../services/network/CognitensorEndpoints';
import DefaultView from '../components/default/DefaultView';
import DashboardHeader from '../components/DashboardHeader';
import DashboardListCard from '../components/DashboardListCard';
import { theme } from '../theme';

const Dashboards = () => {
  const [dashboards, dispatchDashboards] = useReducer(apiStateReducer, {
    data: [],
    isLoading: true,
    isError: false,
  });

  useEffect(() => {
    CognitensorEndpoints.getDashboardList({
      dispatchReducer: dispatchDashboards,
    });
  }, []);

  return (
    <DefaultView>
      <DashboardHeader title="My Dashboards" />
      {dashboards.isError && <Text>Error</Text>}
      {dashboards.isLoading ? (
        <ActivityIndicator size="small" color={theme.colors.primary} />
      ) : (
        <FlatList
          data={dashboards.data.message}
          renderItem={DashboardListCard}
          keyExtractor={(item, index) => index.toString()}
        />
      )}
    </DefaultView>
  );
};

const styles = StyleSheet.create({
  scrollContainer: {
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default Dashboards;
