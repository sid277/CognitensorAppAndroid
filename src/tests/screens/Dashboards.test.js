import React from 'react';
import { shallow } from 'enzyme';
import Dashboards from '../../screens/Dashboards';
import renderer from 'react-test-renderer';


test('should test Dashboards component', () => {
const wrapper = shallow(<Dashboards />);
expect(wrapper).toMatchSnapshot();
});