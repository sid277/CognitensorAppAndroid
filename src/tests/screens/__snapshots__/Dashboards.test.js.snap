// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Dashboards component 1`] = `ShallowWrapper {}`;
